tests/integration/test_envelope.py

```diff
@@ -579,3 +579,51 @@ def test_sample_rates_metrics(mini_sentry, relay_with_processing, events_consume
 
     event, _ = events_consumer.get_event()
     assert event["_metrics"]["sample_rates"] == sample_rates
+
+
+# Checks that we buffer envelopes until we have a global config in processing mode.
+def test_buffer_envelopes_without_global_config(
+    mini_sentry, relay_with_processing, events_consumer
+):
+    include_global = False
+    original_endpoint = mini_sentry.app.view_functions["get_project_config"]
+
+    @mini_sentry.app.endpoint("get_project_config")
+    def get_project_config():
+        nonlocal include_global
+
+        res = original_endpoint().get_json()
+        if not include_global:
+            res.pop("global", None)
+        return res
+
+    mini_sentry.add_basic_project_config(42)
+    events_consumer = events_consumer()
+    options = {"cache": {"global_config_fetch_interval": 1}}
+    relay = relay_with_processing(options=options)
+
+    envelope = Envelope()
+    envelope.add_event({"message": "hello, world!"})
+    relay.send_envelope(42, envelope)
+
+    error_raised = False
+    try:
+        # Event is still in buffer because we have not provided a global config
+        events_consumer.get_event(timeout=1)
+    except AssertionError:
+        error_raised = True
+    assert error_raised
```

Comment:
See [`pytest.raises`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.raises).

It seems a bit fishy that we only wait 1s to check if there are events when there aren't supposed to be, and wait 10s when they are supposed to be. Could we be running into false negatives?

Reply:
hmm, trying but it doesn't seem to work, using it with assertionerror because thats what gets logged in mini_sentry.test_failures:

>[('/api/666/envelope/', AssertionError('Relay sent us event: global config missing in upstream response')), ('/api/666/envelope/', AssertionError('Relay sent us event: global config missing in upstream response')), ('/api/666/envelope/', AssertionError('Relay sent us event: error fetching project states: upstream request returned error 500 Internal Server Error: no error details')), ('/api/666/envelope/', AssertionError('Relay sent us event: global config missing in upstream response'))]

Reply:
Was this answered or addressed ?

Reply:
@olksdr yeah we now wait when asserting empty here